src/main/java/seedu/address/model/person/task/TaskDescription.java

```diff
@@ -0,0 +1,60 @@
+package seedu.address.model.person.task;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a Task's taskDescription in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidTaskDescription(String)}
+ */
+public class TaskDescription {
+    public static final String MESSAGE_CONSTRAINTS = "Task description can take any values, and it should not be blank";
+
+    /*
+     * The first character of the task description must not be a whitespace,
+     * otherwise " " (a blank string) becomes a valid input.
+     */
+    public static final String VALIDATION_REGEX = "[^\\s].*";
+
+    public final String taskDescription;
+
+    /**
+     * Constructs an {@code TaskDescription}.
+     *
+     * @param taskDescription A valid task description.
+     */
+    public TaskDescription(String taskDescription) {
+        requireNonNull(taskDescription);
+        checkArgument(isValidTaskDescription(taskDescription), MESSAGE_CONSTRAINTS);
+        this.taskDescription = taskDescription;
+    }
+
```

Comment:
Maybe can consider trimming the String before checking if it's valid? Because checking like this would mean that strings that start with a whitespace like " Hello" would be invalid

Reply:
hmm that should be done in ParseUtil.parseTaskDescription, the other Person fields are implemented similarly